api/desecapi/tests/test_authentication.py

```diff
@@ -16,15 +16,15 @@ def assertDynDNS12Status(self, status=HTTP_200_OK, authorization=None):
         self.assertStatus(self._get_dyndns12(), status)
 
     def assertDynDNS12AuthenticationStatus(self, username, token, status):
-        # Note that this overwrites self.client's credentials, which may be unexpected
```

Comment:
This note was important as it is likely to change test behavior when running tests together/separately, see also   comment below

Reply:
Ah. I only considered the comment in the context of the following line, and I did not understand why the overwriting would be "unexpected", as the next line does that very explicitly. I therefore thought the comment was leftover from the past.

I now understand that it is unexpected because it happens inside an assertion function, which has a side effect. That's unexpected. `assertDynDNS12AuthenticationStatus` is not used elsewhere, so I'll change it into a local helper function defined inside the test where it's used. This way, all assertions are free of side-effects. Tests are not, but we have the same problem with other tests, so I think as first step that's ok. (We should also make tests free of side-effects.)